src/types.ts

```diff
@@ -27,37 +27,69 @@ export interface Instance {
     account: string,
     bid: Bid
   ) => Promise<boolean>;
-  isZAuctionApprovedToTransferNft: (account: string) => Promise<boolean>;
-  getZAuctionSpendAllowance: (account: string) => Promise<ethers.BigNumber>;
+  isZAuctionApprovedToTransferNft: (
+    account: string,
+    contractAddress: string
+  ) => Promise<boolean>;
+  isZAuctionApprovedToTransferNftLegacy: (
+    account: string,
+    contractAddress: string
+  ) => Promise<boolean>;
   getZAuctionSpendAllowanceByBid: (
     account: string,
     bid: Bid
   ) => Promise<ethers.BigNumber>;
-  getTradeTokenAddress: () => Promise<string>;
+  getZAuctionSpendAllowance: (
+    paymentTokenAddress: string,
+    account: string
+  ) => Promise<ethers.BigNumber>;
+  getZAuctionSpendAllowanceLegacy: (
+    account: string
+  ) => Promise<ethers.BigNumber>;
+  getDefaultPaymentToken: () => Promise<string>;
+  setDefaultPaymentToken: (
```

Comment:
`get/setDefaultPaymentToken` shouldn't be exposed through the SDK ideally. This is code the dApp will never use.